src/OrleansSQLUtils/Storage/RelationalStorage.cs

```diff
@@ -146,11 +146,7 @@ public Task<IEnumerable<TResult>> ReadAsync<TResult>(string query, Action<IDbCom
                 throw new ArgumentNullException("selector");
             }
 
-            //It is certain the result is already ready here and can be collected straight away. Having a truly asynchronous result collection
-            //IAsyncEnumerable<TResult> or equivalent method ought to be used. Taking the result here without async-await saves for generating
-            //the async state machine.
-            var ret = ExecuteAsync(query, parameterProvider, ExecuteReaderAsync, selector, cancellationToken, commandBehavior).GetAwaiter().GetResult().Item1;
```

Comment:
wow, I'm really surprised about this. Looking at the body of `ExecuteAsync` it looks like the comment here is not true. It is not certain that the result is already here. In fact, it's almost guaranteed that the result is not here, and that this code will block the thread.
@veikkoeeva do you agree with this fix, or can you shed some light on why it was like this before?